src/ripple/app/paths/PathRequests.cpp

```diff
@@ -249,7 +249,25 @@ PathRequests::makeLegacyPathRequest(
         app_.getLedgerMaster().newPathRequest();
     }
 
-    return result.second;
+    return std::move (result.second);
+}
+
+Json::Value
+PathRequests::doLegacyPathRequest (
+        Resource::Consumer& consumer,
+        std::shared_ptr<ReadView const> const& inLedger,
+        Json::Value const& request)
+{
+    auto cache = std::make_shared<RippleLineCache> (inLedger);
+
+    auto dummyCompletion = []() { ; };
+    auto req = std::make_shared<PathRequest> (app_, std::move (dummyCompletion),
+        consumer, ++mLastIdentifier, *this, mJournal);
+
+    auto result = req->doCreate (cache, request);
+    if (result.first)
+        result.second = req->doUpdate (cache, false);
```

Comment:
It's possible that we'll call `doUpdate` twice (the first time inside `doCreate` as a fast update, ignoring the result) and then here, as a slow update, saving the result. Is that the intention?